### PR TITLE
✨Enable concurrency in metal3 controllers

### PR DESCRIPTION
**Pull request by kashifest**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

**Comment by mboukhalfa** [comment]
testing the new trigger format
/test-centos-e2e-integration-main

File: main.go [hunk 1]
```diff
@@ -198,33 +206,54 @@ func initFlags(fs *pflag.FlagSet) {
...
 
-	flag.StringVar(
+	fs.StringVar(
 		&healthAddr,
 		"health-addr",
 		":9440",
 		"The address the health endpoint binds to.",
 	)
+
+	fs.IntVar(&metal3MachineConcurrency, "metal3Machine-concurrency", 10,
```
Comment (lentzi90) [inline main.go:237]: In CAPI they seem to [stick to all lower case](https://github.com/kubernetes-sigs/cluster-api/blob/838dcbbb20c45fc5966727151f9ca1320dfcd783/main.go#L163) for the flags, maybe we should do the same? (Same for the rest of the concurrency flags)
```suggestion
	fs.IntVar(&metal3MachineConcurrency, "metal3machine-concurrency", 10,
```

**Review by lentzi90 — COMMENTED** [review]
Thanks! This looks good to me and in line with how CAPI is doing it.
Just one comment below about the flags.

File: main.go (hunk 1 above)
Comment (kashifest) [reply main.go:237]: yes I thought about it and later choose to just go this way. I can change it.

Comment (lentzi90) [reply main.go:237]: I have no strong preference on this but I think it is quite common to keep flags lower case so maybe it is better that way

Comment (tuminoid) [reply main.go:237]: Please do not put upper case characters into flags, I can guarantee nobody can write correctly if you do. :)

Comment (kashifest) [reply main.go]: Done

**Review by lentzi90 — COMMENTED** [review]
/lgtm

**Comment by kashifest** [comment]
@lentzi90 @furkatgofurov7 what should be the icon in the title , ⚠️ or 🌱 ?

**Comment by tuminoid** [comment]
> @lentzi90 @furkatgofurov7 what should be the icon in the title , warning or seedling ?

:sparkles: I think is correct for new features

**Comment by lentzi90** [comment]
:sparkles: makes sense for added feature, but we also change the default from 1 to 10 with this. I don't think such a config change would require :warning: so I would say :sparkles:

**Comment by furkatgofurov7** [comment]
I would say ✨ is an appropriate one IMO as well.

**Review by furkatgofurov7 — APPROVED** [review]
/retitle ✨Enable concurrency in metal3 controllers
/approve
/hold unhold if you think also new feature label is appropriate

**Comment by furkatgofurov7** [comment]
As discussed offline, we will enable the concurrency in BMO, IPAM controllers as well to align all of them.

**Comment by kashifest** [comment]
/hold cancel
I agree with all of you regarding the title

**Pull request merged**